packages/react/src/components/radio-button-group/radio-button-group.tsx

```diff
@@ -139,9 +145,24 @@ export const RadioButtonGroup: VoidFunctionComponent<RadioButtonGroupProps> = ({
     useEffect(() => {
         if (checkedValue !== undefined) {
             setCurrentChecked(checkedValue);
+            lastChecked.current = checkedValue;
         }
     }, [checkedValue]);
```

Comment:
You Might Not Need An Effet™️  😉 
```suggestion
    const [previousCheckedValue, setPreviousCheckedValue] = useState(checkedValue);
    if (checkedValue !== previousCheckedValue) {
        setCurrentChecked(checkedValue);
        setPreviousCheckedValue(checkedValue);
        lastChecked.current = checkedValue;
    }
```